pkg/providers/vsphere/govmomiclient.go

```diff
@@ -0,0 +1,152 @@
+package vsphere
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+type VMOMIFinder interface {
+	Datastore(ctx context.Context, path string) (*object.Datastore, error)
+	Folder(ctx context.Context, path string) (*object.Folder, error)
+	Network(ctx context.Context, path string) (object.NetworkReference, error)
+	ResourcePool(ctx context.Context, path string) (*object.ResourcePool, error)
+	VirtualMachine(ctx context.Context, path string) (*object.VirtualMachine, error)
+}
+
+type VMOMIAuthorizationManager interface {
+	FetchUserPrivilegeOnEntities(ctx context.Context, entities []types.ManagedObjectReference, userName string) ([]types.UserPrivilegeResult, error)
+}
+
+type VSphereClient interface {
```

Comment:
I generate a mnock object off of it for tests in `vsphere_test.go`. https://github.com/aws/eks-anywhere/blob/73b232f33062de067ef0a8668ebb0d2fd6db9386/pkg/providers/vsphere/validator_test.go#L19-L19

Reply:
I am wondering if we are interfacing too much here but I can't think of an alternate solution except for maybe passing in values like datacenter on each validate command so we can potentially remove the need of a builder. We can brainstorm maybe as I want to avoid overcomplicating and it getting stuck in code.